### PR TITLE
Governance votes

### LINKED ISSUE #38: Governance Votes - Silver & Core
We need a silver layer for Governance Votes with a core view. This table will be downstream of the messages model (https://github.com/MetricsDAO/terra_dbt/issues/8 ) and should focus on how individual wallets and validators are voting on governance proposals. 

### To Do 
- [ ]  Assign primary & secondary analysts
- [ ]  Design the schema for this table
- [ ]  Get confirmation on schema
- [ ]  Build schema
- [ ]  Schema build review
- [ ]  Submit Pull Request for review by pod

### Direction 
Governance Votes is a common curated table and many examples can be found in the Flipside data tables, most notably in the [Osmosis database](https://github.com/FlipsideCrypto/osmosis-models/blob/main/models/silver/silver__governance_votes.sql).

The suggested schema includes the following tables and tests:
<html>
<body>

Column | Data Type | Tests | Documentation
-- | -- | -- | --
block_id | number | Unique, not null, sequence gaps, number format | Also known as block height. The block_id, which indicates the length of the blockchain, increases after the addition of each new block.
block_timestamp | timestamp | One day recent, not null, timestamp format | The date and time the block was recorded.
blockchain | string | not null, string/varchar format | The name of the blockchain.
chain_id | string | not null, string/varchar format | The version of the blockchain.
tx_id | string | not null, string/varchar format | A unique key that identifies a transaction. Called "TxHash" on block explorers.
tx_succeeded | boolean | not null | TRUE if the transaction was successful, FALSE otherwise.
voter | string | not null, string/varchar format | The address of the wallet that is casting the vote.
proposal_id | number | not null, number format, no gaps test | An ordered, numerical ID that corresponds to the proposal being voted on. 
vote_option | number | not null, number format | How the user voted on the proposal. 
vote_weight | number | not null, number format | The percentage of their stake that the wallet used to cast their vote. 

</body>
</html>

Note: It would be valuable to provide either a dimensional table that matches the numbers in the vote option column to options ( 1 = 'yes', ...), provide additional documentation that describes this, or to add an additional column in this table containing a text vote option. 

### Contribution Reward
Per the S3 approach, this falls into the project category and contribution incentive will be handled through [Coordinape](https://coordinape.com/). Expected collaboration is 3-4 individuals with a primary Analyst, primary PR Reviewer and possible secondaries for either the Analyst or Review side of the work.

Issue reward pool: $1,200 in LUNA (token amount determined at time of payment)

### PR DESCRIPTION
**Pull request by eevui**
# Description

Exposes governance votes to core views. Closes #38.

# Tests

```
root@07be9b19c28a:/terra# dbt test -s core__fact_governance_votes
16:25:51  Running with dbt=1.2.0
16:25:51  Found 52 models, 262 tests, 0 snapshots, 0 analyses, 599 macros, 2 operations, 0 seed files, 44 sources, 0 exposures, 0 metrics
16:25:51
16:26:28
16:26:28  Running 2 on-run-start hooks
16:26:28  1 of 2 START hook: terra.on-run-start.0 ........................................ [RUN]
16:26:28  1 of 2 OK hook: terra.on-run-start.0 ........................................... [OK in 0.00s]
16:26:28  2 of 2 START hook: terra.on-run-start.1 ........................................ [RUN]
16:26:30  2 of 2 OK hook: terra.on-run-start.1 ........................................... [SUCCESS 1 in 2.43s]
16:26:30
16:26:31  Concurrency: 4 threads (target='dev')
16:26:31
16:26:31  1 of 32 START test accepted_values_core__fact_governance_votes_blockchain__terra  [RUN]
16:26:31  2 of 32 START test accepted_values_core__fact_governance_votes_chain_id__phoenix_1  [RUN]
16:26:31  3 of 32 START test accepted_values_core__fact_governance_votes_vote_option_text__Yes__Abstain__No__NoWithVeto  [RUN]
16:26:31  4 of 32 START test accepted_values_core__fact_governance_votes_vote_weight__1 .. [RUN]
16:26:36  1 of 32 PASS accepted_values_core__fact_governance_votes_blockchain__terra ..... [PASS in 5.26s]
16:26:36  2 of 32 PASS accepted_values_core__fact_governance_votes_chain_id__phoenix_1 ... [PASS in 5.27s]
16:26:36  5 of 32 START test dbt_expectations_expect_column_values_to_be_between_core__fact_governance_votes_vote_option__4__1  [RUN]
16:26:36  6 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes__inserted_timestamp__timestamp_ntz  [RUN]
16:26:36  4 of 32 PASS accepted_values_core__fact_governance_votes_vote_weight__1 ........ [PASS in 5.33s]
16:26:36  7 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_block_id__number  [RUN]
16:26:38  3 of 32 PASS accepted_values_core__fact_governance_votes_vote_option_text__Yes__Abstain__No__NoWithVeto  [PASS in 6.56s]
16:26:38  8 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_block_timestamp__timestamp_ntz  [RUN]
16:26:41  6 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes__inserted_timestamp__timestamp_ntz  [PASS in 4.86s]
16:26:41  9 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_blockchain__string__varchar  [RUN]
16:26:41  7 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_block_id__number  [PASS in 4.81s]
16:26:41  10 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_chain_id__string__varchar  [RUN]
16:26:42  8 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_block_timestamp__timestamp_ntz  [PASS in 4.23s]
16:26:42  11 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_proposal_id__number  [RUN]
16:26:42  5 of 32 PASS dbt_expectations_expect_column_values_to_be_between_core__fact_governance_votes_vote_option__4__1  [PASS in 5.59s]
16:26:42  12 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_tx_id__string__varchar  [RUN]
16:26:46  10 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_chain_id__string__varchar  [PASS in 4.28s]
16:26:46  13 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_tx_succeeded__boolean  [RUN]
16:26:46  9 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_blockchain__string__varchar  [PASS in 4.58s]
16:26:46  14 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_vote_option__number  [RUN]
16:26:47  11 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_proposal_id__number  [PASS in 4.56s]
16:26:47  15 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_vote_option_text__string__varchar  [RUN]
16:26:47  12 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_tx_id__string__varchar  [PASS in 4.63s]
16:26:47  16 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_vote_weight__number  [RUN]
16:26:51  13 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_tx_succeeded__boolean  [PASS in 4.94s]
16:26:51  17 of 32 START test dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_voter__string__varchar  [RUN]
16:26:51  14 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_vote_option__number  [PASS in 4.98s]
16:26:51  18 of 32 START test not_null_core__fact_governance_votes__inserted_timestamp ... [RUN]
16:26:51  15 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_vote_option_text__string__varchar  [PASS in 4.52s]
16:26:51  19 of 32 START test not_null_core__fact_governance_votes_block_id .............. [RUN]
16:26:51  16 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_vote_weight__number  [PASS in 4.60s]
16:26:51  20 of 32 START test not_null_core__fact_governance_votes_block_timestamp ....... [RUN]
16:26:55  18 of 32 PASS not_null_core__fact_governance_votes__inserted_timestamp ......... [PASS in 3.71s]
16:26:55  21 of 32 START test not_null_core__fact_governance_votes_blockchain ............ [RUN]
16:26:55  19 of 32 PASS not_null_core__fact_governance_votes_block_id .................... [PASS in 3.72s]
16:26:55  22 of 32 START test not_null_core__fact_governance_votes_chain_id .............. [RUN]
16:26:55  17 of 32 PASS dbt_expectations_expect_column_values_to_be_in_type_list_core__fact_governance_votes_voter__string__varchar  [PASS in 4.60s]
16:26:55  23 of 32 START test not_null_core__fact_governance_votes_proposal_id ........... [RUN]
16:26:56  20 of 32 PASS not_null_core__fact_governance_votes_block_timestamp ............. [PASS in 4.56s]
16:26:56  24 of 32 START test not_null_core__fact_governance_votes_tx_id ................. [RUN]
16:26:58  22 of 32 PASS not_null_core__fact_governance_votes_chain_id .................... [PASS in 3.38s]
16:26:58  25 of 32 START test not_null_core__fact_governance_votes_tx_succeeded .......... [RUN]
16:26:58  21 of 32 PASS not_null_core__fact_governance_votes_blockchain .................. [PASS in 3.68s]
16:26:58  26 of 32 START test not_null_core__fact_governance_votes_vote_option ........... [RUN]
16:27:00  23 of 32 PASS not_null_core__fact_governance_votes_proposal_id ................. [PASS in 4.66s]
16:27:00  27 of 32 START test not_null_core__fact_governance_votes_vote_option_text ...... [RUN]
16:27:00  24 of 32 PASS not_null_core__fact_governance_votes_tx_id ....................... [PASS in 4.25s]
16:27:00  28 of 32 START test not_null_core__fact_governance_votes_vote_weight ........... [RUN]
16:27:03  26 of 32 PASS not_null_core__fact_governance_votes_vote_option ................. [PASS in 4.41s]
16:27:03  29 of 32 START test not_null_core__fact_governance_votes_voter ................. [RUN]
16:27:03  25 of 32 PASS not_null_core__fact_governance_votes_tx_succeeded ................ [PASS in 4.92s]
16:27:03  30 of 32 START test relationships_core__fact_governance_votes_block_id__block_id__ref_silver__blocks_  [RUN]
16:27:04  28 of 32 PASS not_null_core__fact_governance_votes_vote_weight ................. [PASS in 3.83s]
16:27:04  31 of 32 START test relationships_core__fact_governance_votes_tx_id__tx_id__ref_silver__transactions_  [RUN]
16:27:05  27 of 32 PASS not_null_core__fact_governance_votes_vote_option_text ............ [PASS in 5.20s]
16:27:05  32 of 32 START test unique_core__fact_governance_votes_tx_id ................... [RUN]
16:27:06  29 of 32 PASS not_null_core__fact_governance_votes_voter ....................... [PASS in 3.73s]
16:27:08  30 of 32 PASS relationships_core__fact_governance_votes_block_id__block_id__ref_silver__blocks_  [PASS in 4.91s]
16:27:09  31 of 32 PASS relationships_core__fact_governance_votes_tx_id__tx_id__ref_silver__transactions_  [PASS in 5.00s]
16:27:11  32 of 32 PASS unique_core__fact_governance_votes_tx_id ......................... [PASS in 5.78s]
16:27:11
16:27:11  Finished running 32 tests, 2 hooks in 0 hours 1 minutes and 19.75 seconds (79.75s).
16:27:11
16:27:11  Completed successfully
16:27:11
16:27:11  Done. PASS=32 WARN=0 ERROR=0 SKIP=0 TOTAL=32
```

### CODE REVIEW

**Review by geewynn — APPROVED** [review]

**Comment by geewynn** [comment]
Test passes, code looks good. great job!

LGTM

File: models/core/core__fact_governance_votes.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        block_timestamp,
+        blockchain,
+        chain_id,
+        voter,
+        proposal_id,
+        vote_option,
+        vote_option_text,
+        vote_weight,
+        tx_succeeded,
+        _inserted_timestamp
```
Comment (geewynn) [inline models/core/core__fact_governance_votes.sql:18]: I don't think this column is needed in the core view.

**Review by geewynn — CHANGES_REQUESTED** [review]

File: models/core/core__fact_governance_votes.sql (hunk 1 above)
Comment (eevui) [reply models/core/core__fact_governance_votes.sql:18]: Oh yeah, you're right. Definitely not needed here.

**Review by geewynn — APPROVED** [review]

File: models/core/core__fact_governance_votes.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    select
+        tx_id,
+        block_id,
+        block_timestamp,
+        blockchain,
+        chain_id,
+        voter,
+        proposal_id,
+        vote_option,
+        vote_option_text,
```
Comment (geewynn) [inline models/core/core__fact_governance_votes.sql:15]: Based on the IBC model for governance vote, I think `vote_option_text` should be removed.

Comment (geewynn) [inline models/core/core__fact_governance_votes.sql:15]: Based on the IBC model for governance vote, I think `vote_option_text` should be removed.

**Review by geewynn — CHANGES_REQUESTED** [review]

**Review by geewynn — APPROVED** [review]

**Pull request merged**